src/app/teams/teams-view-finances.component.ts

```diff
@@ -125,11 +125,19 @@ export class TeamsViewFinancesComponent implements OnInit, OnChanges {
           date: [ transaction.date ? new Date(new Date(transaction.date).setHours(0, 0, 0)) : new Date(time), CustomValidators.required ]
         },
         {
-          onSubmit: (newTransaction) => this.submitTransaction(newTransaction, transaction).subscribe(() => {
-            this.planetMessageService.showMessage('Transaction added');
+          onSubmit: (newTransaction) => {
+            if (!transaction._id) {
+              this.submitTransaction(newTransaction, transaction).subscribe(() => {
+              this.planetMessageService.showMessage('Transaction Added');
+              });
+            } else {
+              this.submitTransaction(newTransaction, transaction).subscribe(() => {
+              this.planetMessageService.showMessage('Transaction Edited');
```

Comment:
As a note any time a new block begins indentation should increase by 2 spaces.

After your refactor there should no longer be this level of indentation, but I wanted to note this for future reference.